### ISSUE TITLE
Error occurs when running the Headings tool

### ISSUE THREAD

**Issue by kidokidozh** [microsoft/accessibility-insights-web#4311]
## Describe the bug
Cannot run the "Headings" tool. Our page is having some iframe inside it, not sure whether it's related. Errors output to browser console:
```
injected.bundle.js:77 failed to scan with error - TypeError: Cannot read property 'data' of undefined
injected.bundle.js:77 An axe-core error occurred while processing a result from a child frame.
```

## Screenshots
![image](https://user-images.githubusercontent.com/58833558/120133122-2ade5580-c1fe-11eb-81e7-bb31f3870c7d.png)

## Context (please complete the following information)

- AI-Web Version & Environment: 2.26.1
- Browser Version:
Microsoft Edge Version 91.0.864.37 (Official build) (64-bit)
Google Chrome Version 91.0.4472.77 (Official Build) (64-bit)

**Comment by kidokidozh** [comment]
I found it might be a bug of `axe-core`: https://github.com/dequelabs/axe-core/pull/2965

**Comment by karanbirsingh** [comment]
Thanks @kidokidozh for the report! I agree the issue you found seems like the root cause.

I can reproduce this issue when child iframe descendants have at least 2 headings in total. For instance, the following structures reproduce this issue:

<details>
    <summary>single iframe with two headings</summary>
    
`index.html:`
```html
<html>
    outer page
    <iframe src="page-with-headings.html"></iframe>
</html>
```

`page-with-headings.html:`
```html
<html>
    <h1>heading</h1>
    <h2>heading two</h2>
</html>
```
</details>

<details>
    <summary>iframe with one heading that has a child iframe with another heading</summary>
    
`index.html:`
```html
<html>
    outer page
    <iframe src="page-with-headings.html"></iframe>
</html>
```

`page-with-headings.html:`
```html
<html>
    <h1>heading</h1>
    <iframe src="another-page-with-headings.html"></iframe>
</html>
```

`another-page-with-headings.html:`
```html
<html>
    <h1>heading</h1>
</html>
```
</details>

Does your page match one of these structures?

user impact:
- The headings ad-hoc visualization gets stuck. Users are unable to trigger other actions until refreshing or switching the target page.
- No impact on automated checks in any of our products because we exclude `heading-order` (tagged as best-practice)
- No impact on headings in assessment because we have a custom rule.

We could explore switching from `heading-order` to `collect-headings` (custom rule in assessment) as a workaround. We will look into this a little further in hopes of integrating the external fix or working around it on our side. Thanks!

**Comment by karanbirsingh** [comment]
Fixed in #4327 and released to canary https://chrome.google.com/webstore/detail/accessibility-insights-fo/hbcplehnakffdldhldncjlnbpfgogbem/related
